footer development

ostpbsk opened the pull request:
footer commit

ostpbsk committed:
footer development

footer commit

viktor-kindrat commented:
Task linked: [CU-86bx85757 зверстати футер](https://app.clickup.com/t/86bx85757)

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+            <div class="nav">
+              <div class="nav-logo">
+                <svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
+                  <image xlink:href="D:\footer project\media\logo.svg" width="100" height="100" />
```
Will not work. Paths must be relative

ostpbsk replied:
changed to realtive

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+          <div class="left-section">
+            <div class="nav">
+              <div class="nav-logo">
+                <svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
```
Export SVG as a file and then set on <img /> tag

ostpbsk replied:
exported as .png

viktor-kindrat commented on public/index.html:
In HTML BEM methodology was not used

ostpbsk replied:
changed it according to BEM

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
-        <h1 class="test-headline">Slidy</h1>
-        <p class="test-paragraph">Залиш свій слід у світі подорожей</p>
-    </div>
+        <div class="container">
```
Bad name for a class. Classes should not be named "container" or "block" or "text" etc. They must be named using BEM methodology in depends of which role of them on website

ostpbsk replied:
i deleted this div

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
     <div class="wrap">
-        <h1 class="test-headline">Slidy</h1>
-        <p class="test-paragraph">Залиш свій слід у світі подорожей</p>
-    </div>
```
Github says that you remove closing of ".wrap" tag

ostpbsk replied:
i added one

viktor-kindrat replied:
можливо це через форматування, спробуй відформатувати так як в оригіналі, 4 пропуски на 1 рівень

viktor-kindrat replied:
![image](https://github.com/Train-of-thoughts/slidy-app/assets/84918090/50f3c92d-354d-4f6d-9911-cc8e7b80ad7c)

впевнись шо в тебе отут в редакторі виставлено 4

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+                  <image xlink:href="D:\footer project\media\logo.svg" width="100" height="100" />
+                </svg>
+              </div>
+              <div class="nav-title">НАВІГАЦІЯ ПО САЙТУ</div>
```
Would be better to use `<h2 />` instead of `<div/>`

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+                <li><a href="#">Головна</a></li>
+              </ul>
+            </div>
+            <div class="social" class="left-section">
```
if you want to specify more than one class to one element write class="one-class the-other-one-class"

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+              <div class="social-title">МИ В СОЦ МЕРЕЖАХ</div>
+              <div class="social-item">
+                <div class="social-icon">
+                  <svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
```
also would be better to use <img> tag and exported file

ostpbsk replied:
resolved

viktor-kindrat commented on public/index.html:
```diff
@@ -8,8 +8,79 @@
+              <div class="social-item">
+                <div class="social-icon">
+                  <svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
+                    <image xlink:href="D:\footer project\media\inst.svg" width="25" height="25" />
```
relative paths required

ostpbsk replied:
all paths are realtive now

viktor-kindrat commented on public/index.html:
also I had not left a comment to all images but it should be changed as a first one

ostpbsk replied:
all images are changed to .png

viktor-kindrat commented on public/styles/coreStyles/style.scss:
core styles in any case can not be changed. wrap's styles also! If you want to declare an scss variables, you can create a file variables.scss near style.scss in coreStyles folder

viktor-kindrat replied:
If you want to add your styles you must to create a folder "footer" in styles directory and there create your scss file

ostpbsk replied:
did footer stules in another file and created variables.scss

viktor-kindrat commented on public/styles/coreStyles/style.scss:
```diff
@@ -1,27 +1,131 @@
-    gap: 15px;
-    flex-direction: column;
-    align-items: center;
-    justify-content: center;
```
all above styles should not be removed

viktor-kindrat replied:
all below styles should be removed

viktor-kindrat reviewed (changes requested):
Code has a lot of critical issues. Merging will broke a website

ostpbsk committed:
footer-BEMfix

fixed footer styles names

ostpbsk committed:
footer-variables-fix

created a variables.scss file

ostpbsk committed:
footer-corestyles-fix

created a new folder in styles for footer styles

ostpbsk committed:
footer-container-div-fix

deleted container class and div

ostpbsk committed:
footer-svgtopng-fix

changed all svg files to png files and created a folder media in footer styles folder

viktor-kindrat commented on public/styles/coreStyles/style.scss:
оцей файлик стягни з гілки main. там він добре написаний, тут наче шось не то

viktor-kindrat commented on public/styles/footer/style.scss:
```diff
@@ -0,0 +1,117 @@
+
+// Footer styles
+.wrap {
+    background-color: $footer-bg-color;
```
коли ти встановлюєш бекграунд для вреп ти встановлюєш його для всього сайту а не тільки для футера. Твій компонент мав бути обгорнутий в .footer клас і йому відповідно задаєш стилі

viktor-kindrat replied:
того вреп взагалі не стилізуєм

viktor-kindrat commented on public/styles/footer/style.scss:
```diff
@@ -0,0 +1,117 @@
+    font-style: normal;
+    display: flex;
+    justify-content: space-between;
+    .block-left-section {
```
відповідно якшо в тебе буде батьківський клас для них .footer то в тебе оцей клас буде мати адекватне ім'я .footer__left-section

viktor-kindrat commented on public/styles/footer/style.scss:
```diff
@@ -0,0 +1,117 @@
+        width: auto;
+      }
+
+  .block-nav {
```
ну і всюди, отут наприклад .footer__nav, це буде з дотриманням БЕМ

viktor-kindrat commented on public/index.html:
```diff
@@ -4,12 +4,66 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Slidy - Залиш свій слід у світі подорожей</title>
-    <link rel="stylesheet" href="./styles/coreStyles/style.css">
```
файлики не відключаєм. треба було здублювати рядок і в здубльованому рядку  міняти назву файлу

viktor-kindrat reviewed (changes requested):
впринципі трохи краще, є ще правки зокрема по БЕМ. Твій "блок" це є .footer, решту все шо всередині це його елементи. і те шо всередині тих елементів теж елементи футера. Тобто в тебе всі класи будуть починатись із .footer__some-name

viktor-kindrat committed:
Formatting fixes

ostpbsk committed:
footer-div-and-formatting-fix

ostpbsk committed:
footer-classnames-closingdiv-corestyles-fix

VitaliiDTS reviewed (approved):
Well done

viktor-kindrat reviewed (approved):
Aproved. Code runs as expected. Issues to codestyle: BEM methodology was not correctly used. Must be fixed in the future

viktor-kindrat committed:
Footer development
